frontend/src/core/user/components/UserMenu.js

```diff
@@ -0,0 +1,208 @@
+/* @flow */
+
+import * as React from 'react';
+import onClickOutside from 'react-onclickoutside';
+import { Localized } from 'fluent-react';
+
+import './UserMenu.css';
+
+import SignOut from './SignOut';
+
+import type { NavigationParams } from 'core/navigation';
+import type { UserState } from 'core/user';
+
+
+type Props = {
+    parameters: NavigationParams,
+    signOut: () => void,
+    user: UserState,
+};
+
+type State = {|
+    visible: boolean,
+|};
+
+
+/**
+ * Renders user menu.
+ */
+export class UserMenuBase extends React.Component<Props, State> {
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            visible: false,
+        };
+    }
+
+    toggleVisibility = () => {
+        this.setState((state) => {
+            return { visible: !state.visible };
+        });
+    }
+
+    // This method is called by the Higher-Order Component `onClickOutside`
+    // when a user clicks outside the user menu.
+    handleClickOutside = () => {
+        this.setState({
+            visible: false,
+        });
+    }
+
+    render() {
+        const { parameters, signOut, user } = this.props;
+
+        return <div className="user-menu">
+            <div
+                className="button selector"
+                onClick={ this.toggleVisibility }
+            >
+                { user.isAuthenticated ?
+                <img src={ user.gravatarURLSmall } alt="" /> :
+                <div className="menu-icon fa fa-bars" />
+                }
+            </div>
+
+            { !this.state.visible ? null :
+            <div className="menu"><ul>
+                { !user.isAuthenticated ? null :
+                <li className="details">
+                    <a href="/profile/">
+                        <img src={ user.gravatarURLBig } alt="User avatar" />
+                        <p className="name">{ user.nameOrEmail }</p>
+                        <p className="email">{ user.email }</p>
+                    </a>
+                </li>
+                }
+
+                { !user.isAuthenticated ? null :
```

Comment:
Why not merge that with the previous block?

Reply:
I got `Adjacent JSX elements Parse Error`, but I see I can fix it by using `React.Fragment`. Thanks!

Reply:
Ah, I should have guessed so. You can either use `React.Fragment` indeed (I've been avoiding the form `<> ... </>` because I find it highly confusing, maybe once it becomes more spread in the React community we could revisit) or you can keep it this way, I think that's fine too as it removes a level of indentation.